configuration/debian/tedge_agent/postinst

```diff
@@ -2,17 +2,35 @@
 
 set -e
 
-# Reenable the services only if systemctl is available
+# Initialize the agent. Skip it when agent is running
 if command -v systemctl >/dev/null; then
-    ### Enable the sm services if the device is connected to c8y cloud
-    if [ -f "/etc/tedge/mosquitto-conf/c8y-bridge.conf" ]; then
-        # start and enable tedge-agent
-        systemctl start tedge-agent.service
-        systemctl enable tedge-agent.service
+    if ! systemctl is-active --quiet tedge-agent; then
+         sudo -u tedge -- tedge_agent --init
     fi
 fi
 
-# Initialize the agent
-sudo -- tedge_agent --init
+if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
+        if deb-systemd-helper debian-installed tedge-agent.service; then
+                # This will only remove masks created by d-s-h on package removal.
+                deb-systemd-helper unmask tedge-agent.service >/dev/null || true
+
+                if deb-systemd-helper --quiet was-enabled tedge-agent.service; then
+                        # Create new symlinks, if any.
+                        deb-systemd-helper enable tedge-agent.service >/dev/null || true
+                fi
+        fi
+
+        # Update the statefile to add new symlinks (if any), which need to be cleaned
+        # up on purge. Also remove old symlinks.
+        deb-systemd-helper update-state tedge-agent.service >/dev/null || true
+fi
 
-#DEBHELPER#
+# Enable the services only if systemctl is available
+if command -v systemctl >/dev/null; then
+    # Enable the service if the device is connected to c8y cloud, but
+    # skip it when agent is not running
+    if [ -f "/etc/tedge/mosquitto-conf/c8y-bridge.conf" ] && ! systemctl is-active --quiet tedge-agent ; then
```

Comment:
Why a condition on Cumulocity here?

```suggestion
    if [ ! systemctl is-active --quiet tedge-agent ; then
```

Reply:
Starting the `tedge-agent` is `tedge connect c8y`'s responsibility. If we don't have this condition, `tedge-agent` will start right after the fresh installation.

Reply:
Good point!